### PR TITLE
Add reporting to notification push

### PR DESCRIPTION
**Pull request by davidfurey**
In-memory only for now.  Also need to add topic registration tracking
to registration end-point

### CODE REVIEW

File: common/src/main/scala/tracking/InMemoryNotificationReportRepository.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    Future.successful(\/.right(()))
+  }
+
+  def getByUuid(uuid: String): Future[RepositoryResult[NotificationReport]] = {
+    Future.successful(db.find(_.notification.uuid == uuid) \/> RepositoryError("Notification report not found"))
+  }
+
+  def getByDateRange(from: DateTime, to: DateTime): Future[RepositoryResult[List[NotificationReport]]] = {
+    Future.successful(\/.right(db.filter({report =>
+      (report.sentTime isAfter from) && (report.sentTime isBefore to)
```
Comment (alexduf) [inline common/src/main/scala/tracking/InMemoryNotificationReportRepository.scala]: I think this filter should be inclusive, as in "from <= report.sentTime <= to "

File: notification/app/services/NotificationConfiguration.scala [hunk 2]
```diff
@@ -32,7 +33,10 @@ final class NotificationConfiguration @Inject()(wsClient: WSClient)(implicit exe
 
   private def notificationHub = notificationHubOR.fold(error => throw new Exception(error.message), identity)
 
-  def notificationHubClient = new NotificationHubClient(notificationHub, wsClient)
+  val notificationReportRepository = new InMemoryNotificationReportRepository
```
Comment (alexduf) [inline notification/app/services/NotificationConfiguration.scala]: If we want to have a state on that object, I would suggest that NotificationConfiguration use the @Singleton tag.
That being said I would prefer our state to be outside of the configuration (and ideally injected)

File: notification/app/services/NotificationConfiguration.scala [hunk 3]
```diff
@@ -32,7 +33,10 @@ final class NotificationConfiguration @Inject()(wsClient: WSClient)(implicit exe
...
+  val topicSubscriptionRepository = new InMemoryTopicSubscriptionsRepository
```
Comment (alexduf) [inline notification/app/services/NotificationConfiguration.scala]: same as above

File: registration/app/services/RegistrationConfiguration.scala [hunk 4]
```diff
@@ -32,7 +33,8 @@ final class RegistrationConfiguration @Inject()(wsClient: WSClient)
 
   private def notificationHub = notificationHubOR.fold(error => throw new Exception(error.message), identity)
 
-  def notificationHubClient = new NotificationHubClient(notificationHub, wsClient)
+  val topicSubscriptionRepository = new InMemoryTopicSubscriptionsRepository
```
Comment (alexduf) [inline registration/app/services/RegistrationConfiguration.scala]: Same again

**Comment by alexduf** [comment]
Following our conversation, the configuration object will be refactored as a separate PR. :+1:

File: common/src/main/scala/tracking/InMemoryNotificationReportRepository.scala (hunk 1 above)
Comment (agulowaty) [inline common/src/main/scala/tracking/InMemoryNotificationReportRepository.scala]: http://joda-time.sourceforge.net/apidocs/org/joda/time/ReadableInterval.html#contains(org.joda.time.ReadableInstant) can be used instead isAfter and isBefore

Comment (davidfurey) [inline common/src/main/scala/tracking/InMemoryNotificationReportRepository.scala]: Thanks for that @agulowaty

**Pull request merged**